operator/controllers/kyma_controller.go

```diff
@@ -133,61 +158,43 @@ func (r *KymaReconciler) ReconcileFromConfigMap(ctx context.Context, req ctrl.Re
 
 	var componentNames []string
 	for _, component := range kymaObj.Spec.Components {
+		configMap, err := r.GetTemplateConfigMapForRelease(ctx, component.Name, release.New)
 		componentNames = append(componentNames, component.Name)
-		configMap, err := r.GetConfigMap(ctx, component.Name)
 		if err != nil {
-			logger.Error(err, "component mapping ConfigMap read error, will not re-queue resource %s", req.NamespacedName.String())
+			logger.Error(err, fmt.Sprintf("could not find template configmap for resource %s and release %s, will not re-queue resource %s", component.Name, release.New, req.NamespacedName.String()))
 			return err
 		}
 
 		componentName := component.Name + "-name"
-		//+ uuid.New().String()
 
 		componentBytes, ok := configMap.Data[component.Name]
 		if !ok {
 			return fmt.Errorf("%s component not found for resource %s", component.Name, namespacedName)
 		}
 
-		parsedTemplate, templateErr := template.New(component.Name).Funcs(template.FuncMap{
-			"installOperation":     func(interface{}) string { return string(progression.KymaProgressionPath) },
-			"installTargetVersion": func(interface{}) string { return progression.New },
-			"installOriginVersion": func(interface{}) string { return progression.Old },
-		}).Parse(componentBytes)
+		componentYaml, templateErr := r.GetTemplatedComponent(componentBytes)
 		if templateErr != nil {
 			return fmt.Errorf("error during config map template parsing %w", templateErr)
 		}
 
-		templatedData := bytes.NewBufferString("")
-		templateExecErr := parsedTemplate.Execute(templatedData, nil)
-		if templateExecErr != nil {
-			return fmt.Errorf("error during config map template execution %w", templateExecErr)
-		}
-
-		componentYaml := make(map[string]interface{})
-		if err := yaml.Unmarshal(templatedData.Bytes(), &componentYaml); err != nil {
-			return fmt.Errorf("error during config map unmarshal %w", err)
-		}
-
-		res := unstructured.Unstructured{}
-		res.SetGroupVersionKind(schema.GroupVersionKind{
-			Group:   componentYaml["group"].(string),
-			Kind:    componentYaml["kind"].(string),
-			Version: componentYaml["version"].(string),
-		})
+		res, err := r.GetUnstructuredResource(ctx, schema.GroupVersionResource{
+			Group:    componentYaml["group"].(string),
+			Resource: componentYaml["resource"].(string),
+			Version:  componentYaml["version"].(string),
+		}, componentName, req.Namespace)
```

Comment:
Maybe we could keep the kubebuilder provided r.client.Get() for standardisation?